docs(api): puts the api ref into three columns

PaulRMellor opened the pull request:
**Documentation**

Updates DocGenerator to put the api reference guide tables into three columns: Property, Type and Description.
Puts mentions of the version a property was introduced to the start of a description. 
Also removes the padding method (not needed or we can adjust through CSS)


Plus typo fix!

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

scholzj reviewed (commented):
The original format seems more readable to me. It seems that in most parts, the description as the longest part gets only one third of the width. But in some tables it for whatever reason overflows outside of the window. Not sure what the motivation was for this, but it does not seem to be an improvement for me.

PaulRMellor commented:
> The original format seems more readable to me. It seems that in most parts, the description as the longest part gets only one third of the width. But in some tables it for whatever reason overflows outside of the window. Not sure what the motivation was for this, but it does not seem to be an improvement for me.

Thanks @jakub. I built using bccutil and the table widths were fine. I tried in asciidoctor (which I think you used) and only the KafkaClusterSpec table looked wrong. Were there any others? This was down to the formatting in the source code, which I fixed in the latest commit. I also added some proportional widths to the columns.

I prefer the type on the same row, but if it's not popular, we can revert. Otherwise, maybe there's more that can be done to improve the presentation.

PaulRMellor commented on crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java:
```diff
@@ -250,7 +241,7 @@ private void addAddedIn(Property property) throws IOException {
 
         if (addedIn != null) {
-            out.append(" Added in Strimzi " + addedIn.value() + ".");
+            out.append(" Added in Strimzi " + addedIn.value() + ". ");
```
I thought it would be better to have that information at the front (added to description). I've removed the space, but it didn't affect the output.

scholzj replied:
Why do you think it is better at the front? It seems like something that is not important enough to be in the first place. Especially after a few releases when it is not new anymore and almost everyone now uses a newer release.

k-wall commented:
I'm an infrequent reader of the strimzi documentation.  To me the three column layout is clearer/more obvious.  With the two column version I think the reader can get lost, especially when looking at tables with more rows.

fvaleri commented:
I definitely vote for the three-columns format, as I always struggled with the current format.

ppatierno reviewed (approved):
LGTM. I prefer three columns.

mimaison commented:
I find the 3 column format easier to read.
